File: fixtures/scale/csharp/medium/Memory/HistoryLog.cs
using System.Collections.Generic;

namespace Calc.Memory
{
    public class HistoryLog
    {
        private readonly List<HistoryEntry> _entries = new List<HistoryEntry>();

        public HistoryEntry Append(string expression, double value)
        {
            var entry = new HistoryEntry(expression, value);
            _entries.Add(entry);
            return entry;
        }

        public List<HistoryEntry> Recent(int count)
        {
            var start = _entries.Count > count ? _entries.Count - count : 0;
            return _entries.GetRange(start, _entries.Count - start);
        }

        public int Count()
        {
            return _entries.Count;
        }
    }
}

File: fixtures/scale/swift/medium/Memory/HistoryLog.swift
final class HistoryLog {
    private var entries: [HistoryEntry] = []

    @discardableResult
    func append(_ expression: String, _ value: Double) -> HistoryEntry {
        let entry = HistoryEntry(expression: expression, value: value, stamp: Date())
        entries.append(entry)
        return entry
    }

    func recent(_ count: Int) -> [HistoryEntry] {
        let start = entries.count > count ? entries.count - count : 0
        return Array(entries[start...])
    }

    func count() -> Int {
        return entries.count
    }
}
